projects/extension/src/background/ConnectionManager.test.ts

```diff
@@ -237,46 +237,6 @@ describe('When the manager is shutdown', () => {
   });
 });
 
-describe('Check storage and send notification when adding an app', () => {
-  const manager = new ConnectionManager();
-
-  chrome.storage.sync.get.mockImplementation((keys, callback) => {
-    callback({ notifications: true }) 
-  });
-
-  beforeEach(async () => {
-    chrome.storage.sync.get.mockClear();
-    chrome.notifications.create.mockClear();
-    manager.smoldotLogLevel = 1;
-    await manager.initSmoldot();
-  });
-
-  afterEach( () => {
-    manager.shutdown();
-  })
-
-  test('Checks storage for notifications preferences', () => {
```

Comment:
We should reinstate these 2 tests.  You'll need to spy on addChain to trigger the callback

Reply:
The reason why these were removed is that the notification was added inside the private `#handleSpecMessage` in order to be initiated on correct timing and have the correct information in it.This is triggered through the port of the specific App. In all of our tests we are [mocking the port](https://github.com/paritytech/substrate-connect/blob/69eb7b8c02ff89e8681b481d52fdae10a988ac31/projects/extension/src/background/ConnectionManager.test.ts#L24) - so I am not really sure how the spy on addChain would help. Notifications could be tested with Cypress on a more end to end test suite (instead of a unit one) probably.

Reply:
You can add a spy on `ConnectionManager.addChain` on your instance of the `ConnectionManager` just before sending the spec message using your `MockPort` and configure your spy to return a resolved promise then you can check that the mock versions of the chrome API were called appropriately.  This will test the logic in isolation and run much faster than a fullstack test